### PR TITLE
Support multiple OpenAPI json in one crawler

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

As title

### 🤓 What?

- Refactor the extraction
- Support multiple OAS in config

### 🧪 Tested?

Unit test, and verified locally

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/openapi/config.py [hunk 1]
```diff
@@ -20,13 +20,36 @@ class OpenAPIAuthConfig:
...
         return self
+
+
+@dataclass(config=ConnectorConfig)
+class OpenAPIRunConfig(BaseConfig):
+    base_url: Optional[HttpUrl] = None
+    openapi_json_path: Optional[FilePath] = None
+    openapi_json_url: Optional[HttpUrl] = None
+    auth: Optional[OpenAPIAuthConfig] = None
+    specs: Optional[List[OpenAPIJsonConfig]] = None
```
Comment (usefulalgorithm) [inline metaphor/openapi/config.py:41]:
```suggestion
    specs: List[OpenAPIJsonConfig]] = field(default_factory=list)
```
And then add a validator to ensure `specs` isn't empty?

File: metaphor/openapi/extractor.py [hunk 2]
```diff
@@ -44,51 +44,71 @@ def from_config_file(config_file: str) -> "OpenAPIExtractor":
...
+            [
+                OpenAPIJsonConfig(
+                    base_url=config.base_url,
+                    openapi_json_path=config.openapi_json_path,
+                    openapi_json_url=config.openapi_json_url,
+                )
+            ]
+            if config.base_url
+            else []
+        ) + (config.specs or [])
```
Comment (usefulalgorithm) [inline metaphor/openapi/extractor.py:91]: and then we won't need this

**Review by usefulalgorithm — COMMENTED** [review]
Or why not just have multiple OpenAPI crawlers?

File: metaphor/openapi/config.py (hunk 1 above)
Comment (elic-eon) [reply metaphor/openapi/config.py:41]: We try not to introduce a breaking change, we can use your suggestion when we bump to 0.15

**Review by usefulalgorithm — APPROVED** [review]

**Pull request merged**